internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2651,3 +2704,66 @@ func flattenKubernetesClusterMaintenanceConfigurationTimeInWeeks(input *[]contai
 	}
 	return results
 }
+
+func expandKubernetesClusterHttpProxyConfig(input []interface{}) *containerservice.ManagedClusterHTTPProxyConfig {
+	httpProxyConfig := containerservice.ManagedClusterHTTPProxyConfig{}
+	if len(input) == 0 {
+		return nil
+	}
+
+	config := input[0].(map[string]interface{})
+	httpProxyConfig.HTTPProxy = utils.String(config["http_proxy"].(string))
+	httpProxyConfig.HTTPSProxy = utils.String(config["https_proxy"].(string))
+	httpProxyConfig.TrustedCa = utils.String(config["trusted_ca"].(string))
+	noProxyRaw := config["no_proxy"].([]interface{})
+	httpProxyConfig.NoProxy = utils.ExpandStringSlice(noProxyRaw)
+	return &httpProxyConfig
+}
+
+func flattenKubernetesClusterHttpProxyConfig(props *containerservice.ManagedClusterProperties, flattenedDefaultNodePool *[]interface{}) []interface{} {
+	results := make(map[string]interface{})
+	httpProxyConfig := props.HTTPProxyConfig
+	if httpProxyConfig != nil {
+		results["http_proxy"] = httpProxyConfig.HTTPProxy
+		results["https_proxy"] = httpProxyConfig.HTTPSProxy
+		noProxyList := httpProxyConfig.NoProxy
+		// Removing the default values that AKS always applies from the flatten
```

Comment:
whilst this is trying to be helpful, the user should specify all of these values since this is cluster configuration related - so can we remove this logic

Reply:
Thank you for all of the feedback. 

If I do not include this ignore no_proxy logic, I am running into some issues with the subsequent runs because the list is populated with (even if no value is passed into no_proxy):
no_proxy    = [
  "localhost",
  "127.0.0.1",
  "168.63.129.16",
  "169.254.169.254",
  "konnectivity",
  "akstestrxaxlbiw-1c9d9635.hcp.eastus.azmk8s.io",
  "10.240.0.0/12",
  "10.244.0.0/16",
  "10.0.0.0/16",
  "172.17.0.1/16",
]

Specifically the akstestrxaxlbiw-1c9d9635.hcp.eastus.azmk8s.io has 1c9d9635 which you would need to add after the initial run since that value is seemingly randomly generated. 

Also, the order does not seem to be preserved so removing this logic gives me this error with the tests:
 Step 1/2 error: After applying this test step, the plan was not empty.

Please let me know any ideas to proceed on this part.

Reply:
I'm curious how @tombuildsstuff would suggest to solve it, but I'd consider a `ForceNew` `no_proxy_user_managed ` list and a computed `no_proxy_computed` list for simplicity. The computed values will change in the future, `azurerm` should have nothing to do with it.

I'm not completely sure how that would work on updates because I don't know what the API expects here (full list or only `user_managed`). If only `user_managed` would be fine then it is as easy as that, otherwise you'd have to merge them somehow.

I'm sure my naming suggestions could be improved 😅

Reply:
So I was able to get past my test issues (plan was not empty.) by adding:
lifecycle {
    ignore_changes = [
      http_proxy_config.0.no_proxy
    ]
  }
  
Not sure if that is really ideal or not.